tmux-cc/src/lib.rs

```diff
@@ -489,30 +489,42 @@ impl Parser {
         }
     }
 
-    pub fn advance_byte(&mut self, c: u8) -> Option<Event> {
+    pub fn advance_byte(&mut self, c: u8) -> anyhow::Result<Option<Event>> {
         if c == b'\n' {
             self.process_line()
         } else {
             self.buffer.push(c);
-            None
+            Ok(None)
         }
     }
 
-    pub fn advance_string(&mut self, s: &str) -> Vec<Event> {
+    pub fn advance_string(&mut self, s: &str) -> anyhow::Result<Vec<Event>> {
         self.advance_bytes(s.as_bytes())
     }
 
-    pub fn advance_bytes(&mut self, bytes: &[u8]) -> Vec<Event> {
+    pub fn advance_bytes(&mut self, bytes: &[u8]) -> anyhow::Result<Vec<Event>> {
         let mut events = vec![];
-        for &b in bytes {
-            if let Some(event) = self.advance_byte(b) {
-                events.push(event);
+        for (i, &b) in bytes.iter().enumerate() {
+            match self.advance_byte(b) {
+                Ok(option_event) => {
+                    if let Some(e) = option_event {
+                        events.push(e);
+                    }
+                }
+                Err(err) => {
+                    // TODO: concat remained bytes
+                    return Err(anyhow::anyhow!(format!(
+                        "{}{}",
+                        err,
+                        std::str::from_utf8(&bytes[i..]).unwrap()
```

Comment:
prefer https://doc.rust-lang.org/std/string/struct.String.html#method.from_utf8_lossy so that we can avoid the unwrap in case we have non-utf8 input, and still be able to display a reasonable error message without crashing all of the users panes!